src/lib/Settings.svelte

```diff
@@ -1,32 +1,69 @@
 <script lang="ts">
     import {createEventDispatcher, onMount} from "svelte";
-
-    export let scannerPath: string
-    export let donePath: string
+    import {invoke} from "@tauri-apps/api";
+    import {settings} from "./util/settings";
 
     const dispatch = createEventDispatcher()
-    let tmpScannerPath: string
-    let tmpDonePath: string
+    let scannerPath: string
+    let donePath: string
 
-    onMount(() => {
-        tmpScannerPath = scannerPath;
-        tmpDonePath = donePath;
+    onMount(async () => {
+        scannerPath = await settings.scannerPath;
+        donePath = await settings.donePath;
     })
 
-    function dispatchSave() {
-        dispatch("save", {
-            newScanPath: tmpScannerPath,
-            newDonePath: tmpDonePath
-        })
+    async function pickScannerPath() {
+        await invoke<string>('pick_directory', {startPath: scannerPath})
+            .then(path => {
+                scannerPath = path
+                dispatch("save", {newScanPath: path})
+            })
+            .catch(error => console.log(error))
     }
-</script>
 
-<form on:submit|preventDefault={() => dispatchSave()}>
-    <label for="scannerPath">Skanner kilde</label>
-    <input type="text" id="scannerPath" bind:value={tmpScannerPath}/>
+    async function pickDonePath() {
+        await invoke<string>('pick_directory', {startPath: donePath})
+            .then(path => {
+                donePath = path
+                dispatch("save", {newDonePath: path})
+            })
+            .catch(error => console.log(error))
+    }
 
-    <label for="donePath">Mappe for ferdige objekter</label>
-    <input type="text" id="donePath" bind:value={tmpDonePath}/>
+</script>
 
-    <button type="submit">Save</button>
+<form on:submit|preventDefault class="settings-form">
+    <div class="form-group">
+        <label for="scannerPath">Skanner kilde</label>
+        <button on:click={() => pickScannerPath()}>Velg mappe</button>
+        <input disabled type="text" id="scannerPath" bind:value={scannerPath}/>
+    </div>
+
+    <div class="form-group">
+        <label for="donePath">Ferdige objekter</label>
+        <button on:click={() => pickDonePath()}>Velg mappe</button>
+        <input disabled type="text" id="donePath" bind:value={donePath}/>
```

Comment:
Ser text inputten er disabled. Lurer på om det er greit å åpne for redigering rett i feltet her også. Om man vil paste inn en path som er standard på en seksjon.

Reply:
Åpnet for å skrive manuelt, la også til sjekk at det faktisk er en mappe og synlig vise at det blir lagret.